packages/cli/test/adapter-lifecycle.test.ts

```diff
@@ -0,0 +1,334 @@
+            const listResult = await runCliCommand(['list', 'adapters']);
+            expect(listResult.exitCode).to.equal(0);
+
+            if (!listResult.stdout.includes(testAdapter)) {
+                // Try to install the adapter
+                const installResult = await runCliCommand(['install', testAdapter], 60000);
+                
+                if (installResult.exitCode !== 0) {
+                    if (installResult.stderr.includes('TIMEOUT') || 
+                        installResult.stderr.includes('getaddrinfo ENOTFOUND') ||
+                        installResult.stderr.includes('dns block')) {
+                        console.log(`Skipping test - network not available for adapter installation`);
+                        this.skip();
+                        return;
+                    }
+                }
+                
+                expect(installResult.exitCode).to.equal(0, `Failed to install ${testAdapter}: ${installResult.stderr}`);
+                adapterWasInstalled = true;
+                
+                // Verify the adapter was actually installed
+                const verifyResult = await runCliCommand(['list', 'adapters']);
+                expect(verifyResult.exitCode).to.equal(0);
+                expect(verifyResult.stdout).to.include(testAdapter);
+                
+                // Check that adapter files were created
+                const adapterDir = path.join(testDir, 'node_modules', `iobroker.${testAdapter}`);
+                expect(await fs.pathExists(adapterDir)).to.be.true;
+                
+                // Check that io-package.json exists
+                const ioPackageFile = path.join(adapterDir, 'io-package.json');
+                expect(await fs.pathExists(ioPackageFile)).to.be.true;
+                
+                // Validate the io-package.json structure
+                const ioPackage = await fs.readJSON(ioPackageFile);
+                expect(ioPackage).to.have.property('common');
+                expect(ioPackage.common).to.have.property('name', testAdapter);
+            } else {
+                console.log(`${testAdapter} adapter already installed`);
+            }
+        });
+
+        it('should list adapters and show installed adapters', async function () {
+            const result = await runCliCommand(['list', 'adapters']);
+
+            expect(result.exitCode).to.equal(0);
+            // Should not timeout or crash
+            expect(result.stderr).to.not.include('TIMEOUT');
+        });
+
+        it('should create an adapter instance and validate objects/states', async function () {
+            // First check if adapter exists
+            const listResult = await runCliCommand(['list', 'adapters']);
+
+            if (listResult.stdout.includes(testAdapter)) {
+                // Try to create an instance
+                const result = await runCliCommand(['add', testAdapter, '--enabled', 'false']);
```

Comment:
verify if and when yes how many instaled are alreayd there. This command is then adding a new instance